### DISCUSSION TITLE
Difference - "pyenv uninstall" and "pyenv virtualenv-delete"

### DISCUSSION THREAD

**Discussion by baskrahmer** [pyenv/pyenv#2534] [Q&A] [answered]
When I run `pyenv versions`, I get to see all my versions. As far as I believe, they are all unique. Example:

```
$ pyenv versions
  system
  3.9.0
  3.9.0/envs/test
```

When I want to remove a version, I can use `pyenv uninstall 3.9.0`. For the venv however, I need to run `pyenv virtualenv-delete 3.9.0/envs/test`. Does this really need to be a separate command? I think it complicates things.

**Comment by baskrahmer** [comment]
Pyenv 2.3.4 by the way :)

**Comment by native-api** [comment]
It could be the same command, but I'm not sure if that would really simplify things.
* You create a virtualenv in a different way than installing a new Python version.
* It's reflected in the system and accessed in a different way, too.

So why should it suddenly uninstall the same way?
If anything, that runs the risk of accidentally uninstalling a full Python installation instead of a single environment.

**Reply by baskrahmer** [reply]
I see, those are good arguments. I guess that means when you uninstall a Python version, it would then also remove all virtualenvs built on top of that, meaning the adverse effect of uninstalling a full Python version are greater.

From my perspective though; I don't use pyenv daily and just found it a bit confusing how to clean up my installations/virtualenvs. This SO post might contribute to that confusion: https://stackoverflow.com/questions/65097575/how-do-i-remove-pyenv-virtualenvs
In the post it is stated to use `pyenv uninstall` for venvs (which was my initial guess as well).

**Reply by native-api** [reply]
Well, you can create a feature request and see what feedback it gets.

**Comment by native-api** [comment, marked as answer]
`pyenv uninstall` now accepts virtualenv names:

```
$ pyenv virtualenv 3 test
$ pyenv versions
* system (set by /home/vmuser/.pyenv/version)
  3.14.0
  3.14.0/envs/test
  test --> /home/vmuser/.pyenv/versions/3.14.0/envs/test
$ pyenv uninstall test
pyenv: remove /home/vmuser/.pyenv/versions/test? (y/N) y
pyenv-virtualenv: remove /home/vmuser/.pyenv/versions/3.14.0/envs/test? (y/N) y
$ pyenv virtualenv 3 test
$ pyenv uninstall 3.14.0/envs/test 
pyenv: remove /home/vmuser/.pyenv/versions/test? (y/N) y
pyenv-virtualenv: remove /home/vmuser/.pyenv/versions/3.14.0/envs/test? (y/N) y
```
